warehouse/query-core/src/main/java/datawave/query/predicate/EventDataQueryFieldFilter.java

```diff
@@ -57,7 +62,10 @@ public void startNewDocument(Key document) {
      */
     @Override
     public boolean keep(Key k) {
-        return true;
+        DatawaveKey key = new DatawaveKey(k);
+        
+        // keep as long as it is part of the event or it is part of the query
```

Comment:
So apply means add to document, keep means send back to user (but only those actually added to the document).
This says we are sending back to the user anything we add to the document.  Is that what we want?

Reply:
If this is dead code then just return true.

Reply:
set in next commit